### DISCUSSION TITLE
cobbler needing --batch functionality added back in, causing errors detecting named vs named-chroot.

### DISCUSSION THREAD

**Discussion by cinnion** [cobbler/cobbler#3843] [Q&A] [answered]
Guten abend, alles!

In getting things set up, I have a long batch file to import a cleaned up configuration into cobbler. Surely, I am not the only person who has done bulk adds/edits. Because everytime we edit a system, we resync and restart named, this really quickly results in systemctl flagging named-chroot as restarting too rapidly, and then things quickly devolve into trying to restart named instead.  This was not an issue when the resync/restart had to be explicitly triggered by a `cobbler sync`, but now, at least on RL 9, it results in changes to the `/var/named` tree which require manual fixing.

On workaround would be to add sleeps between each cobbler command in the shell script, but this is sub-optimal at best. Failing that, there are two possible fixes beyond the non-starter of going back to requiring a sync being manually requested.  They are:

1) Recover the ancient functionally of disabling the sync/restart by specifying a flag on the commands. IIRC, this flag was `--batch`.
2) Rather than detecting `named` vs. `named-chroot` dynamically, make it a config option in say settings.yaml.

BTW... the threshold is really low to have systemctl think that the service is being restarted too quickly... like only 3-5 times in less than a second, which is easily reachable at least early on in a bulk script.

Thoughts?

**Comment by SchoolGuy** [comment, marked as answer]
We are just about to re-enable the support for transactions (see #3814). Once it is merged, I will backport it (as it is a small change). I am aware of the issue and hit it during my testing of the new Golang-based CLI too.

**Reply by cinnion** [reply]
Thanks!  And go? Sounds even more like I better pop that language to the top of my learning list.
